pkg/node/devnode.go

```diff
@@ -301,7 +302,8 @@ func NewDevBee(logger log.Logger, o *DevOptions) (b *DevBee, err error) {
 				}
 
 				post.HandleDepthIncrease(batch.ID, newDepth)
-				return nil
+
+				return mockPost.ErrNotImplemented
```

Comment:
We should ONLY return this error. Why do we need L288-L304?

Same for the above function. These functions should just be 

```
func(ctx context.Context, batchID []byte, newDepth uint8) error {
        return ErrNotImplemented
}
```

Also, the error should not be defined in the mock package. Mock package is not expected to implement things, so housing the NotImplemented error there doesnt make sense. There could possible be a ErrNotImplemented already defined somewhere. If not you should ideally define it in the `node` package.